ratatui-widgets/src/mascot.rs

```diff
@@ -0,0 +1,196 @@
+               ████████████████
+                ███ ██████████
+              ▒▒    █████████
+             ▒░░▒   █████████
+            ▒░░░░▒ ██████████
+           ▒░░▓░░░▒ █████████
+          ▒░░▓▓░░░░▒ ████████
+         ▒░░░░░░░░░░▒ ██████████
+        ▒░░░░░░░░░░░░▒ ██████████
+       ▒░░░░░░░▓▓░░░░░▒ █████████
+      ▒░░░░░░░░░▓▓░░░░░▒ ████  ███
+     ▒░░░░░░░░░░░░░░░░░░▒ ██   ███
+    ▒░░░░░░░░░░░░░░░░░░░░▒ █   ███
+    ▒░░░░░░░░░░░░░░░░░░░░░▒   ███
+     ▒░░░░░░░░░░░░░░░░░░░░░▒ ███
+      ▒░░░░░░░░░░░░░░░░░░░░░▒ █"
+};
+
+const RAT: char = '█';
+const HAT: char = 'h';
+const EYE: char = 'e';
+const TERM: char = '░';
+const TERM_BORDER: char = '▒';
+const TERM_CURSOR: char = '▓';
+
+/// State for the mascot's eye
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum MascotEye {
+    /// The default eye color
+    #[default]
+    Default,
+
+    /// The red eye color
+    Red,
+}
+
+/// A widget that renders the Ratatui mascot
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct RatatuiMascot {
+    eye_state: MascotEye,
+    /// The color of the rat
+    rat_color: Color,
+    /// The color of the rat's eye
+    rat_eye_color: Color,
+    /// The color of the rat's eye when blinking
+    rat_eye_blink: Color,
+    /// The color of the rat's hat
+    hat_color: Color,
+    /// The color of the terminal
+    term_color: Color,
+    /// The color of the terminal border
+    term_border_color: Color,
+    /// The color of the terminal cursor
+    term_cursor_color: Color,
+}
+
+impl Default for RatatuiMascot {
+    fn default() -> Self {
+        Self {
+            rat_color: Color::Indexed(252),         // light_gray #d0d0d0
+            hat_color: Color::Indexed(231),         // white #ffffff
+            rat_eye_color: Color::Indexed(236),     // dark_charcoal #303030
+            rat_eye_blink: Color::Indexed(196),     // red #ff0000
+            term_color: Color::Indexed(232),        // vampire_black #080808
+            term_border_color: Color::Indexed(244), // gray  #808080
+            term_cursor_color: Color::Indexed(248), // dark_gray #a8a8a8
+            eye_state: MascotEye::Default,
+        }
+    }
+}
+
+impl RatatuiMascot {
+    /// Create a new Ratatui mascot widget
+    pub fn new() -> Self {
+        Self {
+            ..Default::default()
+        }
+    }
+
+    /// Set the eye state (open / blinking)
+    #[must_use]
+    pub const fn set_eye(self, rat_eye: MascotEye) -> Self {
+        Self {
+            eye_state: rat_eye,
+            ..self
+        }
+    }
+
+    const fn color_for(&self, c: char) -> Option<Color> {
+        match c {
+            RAT => Some(self.rat_color),
+            HAT => Some(self.hat_color),
+            EYE => Some(match self.eye_state {
+                MascotEye::Default => self.rat_eye_color,
+                MascotEye::Red => self.rat_eye_blink,
+            }),
+            TERM => Some(Color::Reset), // use the terminal color
```

Comment:
I missed seeing this before. This is definitely cool (and might make a good option for the widget), but I think the default of the darker known color would be generally better here than making this transparent to the underlying terminal color. Mainly because it makes the demo render consistently on any terminal.